### DISCUSSION TITLE
Feature: support different languages

### DISCUSSION THREAD

**Discussion by hahhen** [mdx-editor/editor#324] [Q&A] [answered]
I would gladly try to translate the editor to my mother language, is it possible? Or are there already different languages for the editor?

**Comment by petyosi** [comment, marked as answer]
The editor does not have an i18n capability at the moment. I don't have an up-to-date knowledge on the recommended ways to make a reusable component multilingual, given that different apps may have different tools for the purpose. Are you aware of any good examples of such components?

**Reply by hahhen** [reply]
This video shows how to integrate i18next on a React project:
https://www.youtube.com/watch?v=3BPM_M4gGso

Do you believe that could work?

In the video, he codes the translations directly inside resources, but you can also create a folder such as locales and include different JSON files for each language.
I'm not sure if this is the best option for MDXEditor specifically, but it could work, I believe.

Apart from this video, there's another editor called ByteMD that also has i18n, but I'm not sure how it works exactly, but since you are a more experienced developer, perhaps you can understand it better
https://github.com/bytedance/bytemd/tree/main/packages/bytemd/locales

**Reply by hahhen** [reply]
Good evening, petyosi! I see that you've implemented the i18n feature, I wanted to say thank you so much and congratulations! You've managed to build THE BEST markdown wyiswyg editor out there! I'm also so happy to see your project grow, like the app.mdxeditor that's being developed.

I also wanted to ask: is there a way to send one-time donations below $100? I wanted to support the project, but the minimum Github $100 amount is a lot of money for me ~v~, I'd send a donation right away!

**Reply by petyosi** [reply]
@hahhen - thank you for the kind words. I've set a minimum amount of donations because previously people were sending me amounts like $1 (one dollar), and then asking me for priority support and feature development. 

If $100 is too much for you at the moment, I understand that. You can contribute in other ways, and in general, use the editor as much as you want, and spread the word about it. I hope your projects grow successful and profitable. At some point, your business can sponsor my work as a means of appreciation.

**Reply by hahhen** [reply]
No problem! Thank you very much!

**Comment by Prominence** [comment]
I vote for the topic. I need that functionality in my project. 
Probably you could add callbacks implementation. I believe I saw something similar in your documentation, but I could not find it.

Example: 
```
<MDXEditor
    ...
    plugins={[
        somePlugin({ title: code => i18n.translate(code) }),
    ]}
    ...
/>
```

**Reply by NicolasKulka** [reply]
Need change language of wysiwyg
